src/computability/tm_to_partrec.lean

```diff
@@ -763,6 +770,12 @@ inductive Λ'
 
 instance : inhabited Λ' := ⟨Λ'.ret cont'.halt⟩
 
+instance : decidable_eq Λ' :=
+λ a b, begin
+  induction a generalizing b; cases b; try { apply decidable.is_false, rintro ⟨⟨⟩⟩, done },
+  all_goals { exactI decidable_of_iff' _ (by simp [function.funext_iff]) },
+end
```

Comment:
Is this something that the `derive_handler` should be doing automatically?

Reply:
The reason it was implemented manually is because one of the variants uses a function type over a finite set, and the derive handler doesn't know how to handle that. (It's in core, so it doesn't even know what finiteness is.)

Reply:
Could the derive handler just ask for a `decidable_eq (finset α → β)` instance through typeclass resolution, which presumably mathlib provides?